src/ert/gui/model/node.py

```diff
@@ -9,28 +17,128 @@ class NodeType(Enum):
     JOB = auto()
 
 
-class Node:
-    def __init__(self, id_: int, data: Dict, type_: NodeType) -> None:
-        self.parent: Optional[Node] = None
-        self.data: Dict = data
-        self.children: Dict[int, Node] = {}
-        self.id = id_
-        self.type = type_
+@dataclass
+class _Node(ABC):
```

Comment:
Is there a particular reason why `Node` is renamed to `_Node`?

Reply:
I wanted it as a private class, and it should only be used for inheriting in the same module.
Is there a better way to do this, or should I remove the prefix _?

Reply:
The prefix is only a social thing anyways, there is nothing in Python stopping you from doing it still. I am not sure if it is really important to us that nothing else is subclassing it, if so it would probably be inside Ert in any case so if it happens, it might be warranted... This a very non-important detail :)